domains/Discussions/Tests/Feature/QuestionsUpdateTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Domains\Discussions\Tests\Feature;
+
+use Domains\Accounts\Database\Factories\UserFactory;
+use Domains\Accounts\Models\User;
+use Domains\Discussions\Database\Factories\QuestionFactory;
+use Domains\Discussions\Models\Question;
+use Faker\Factory;
+use Faker\Generator;
+use Illuminate\Http\Response;
+use Illuminate\Support\Carbon;
+use Illuminate\Support\Str;
+use Laravel\Lumen\Testing\DatabaseMigrations;
+use Tests\TestCase;
+
+class QuestionsUpdateTest extends TestCase
+{
+    use DatabaseMigrations;
+
+    private Generator $faker;
+    private User $user;
+    private Question $question;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->faker = Factory::create();
+        $this->user = UserFactory::new()->create();
+        $this->question = QuestionFactory::new(['author_id' => $this->user->id])->create();
+    }
+
+    /** @test */
+    public function it_updates_questions(): void
+    {
+        $payload = [
+            'title' => $this->faker->title,
+            'description' => $this->faker->paragraph,
+        ];
+
+        $this->actingAs($this->user);
+        $response = $this->call(
+            'PATCH',
+            route('discussions.questions.update', ['questionId' => $this->question->id]),
+            $payload
+        );
+        self::assertTrue($response->isEmpty());
+
+        $this->seeInDatabase('questions', [
+            'author_id' => $this->user->id,
+            'title' => $payload['title'],
+            'slug' => Str::slug($payload['title']),
+            'description' => $payload['description'],
+            'updated_at' => Carbon::now()
```

Comment:
This is tricky. If for some reason the execution time of when the resource was updated and the time when this assertion is being executed differ, this test fails.

Believe me, I've seen too many intermittent failing tests due to these types of things. We need to freeze time before the test starts, with a call to `Carbon::setTestNow()` :stuck_out_tongue:

Reply:
I actually thought about this but figured: nah, won't happen :)
Will correct.